### PR TITLE
Support rendering to custom framebuffer

### PR DESCRIPTION
**Pull request by Pessimistress**
#### Change List
- Add experimental `_framebuffer` prop to `Deck`
- Add test app
- Documentation

### CODE REVIEW

**Comment by tsherif** [comment]
Could this not be handled by the application just binding a framebuffer in the `onBeforeRender` callback?

**Comment by Pessimistress** [comment]
@tsherif I think any user setting will be overwritten by https://github.com/uber/deck.gl/blob/master/modules/core/src/passes/layers-pass.js#L9

File: modules/core/src/lib/deck-renderer.js [hunk 1]
```diff
@@ -46,16 +47,14 @@ export default class DeckRenderer {
 
     opts.layerFilter = this.layerFilter;
     opts.effects = opts.effects || [];
+    opts.target = opts.target || Framebuffer.getDefaultFramebuffer(this.gl);
```
Comment (ibgreen) [inline modules/core/src/lib/deck-renderer.js]: Nit: In the spirit of consistency / typing things out and making them self-documenting, should we call this `options.framebuffer` or `options.targetFramebuffer`?

File: modules/core/src/lib/deck-renderer.js [hunk 2]
```diff
@@ -83,11 +82,11 @@ export default class DeckRenderer {
   }
 
   // Private
-  _preRender(effects, params) {
+  _preRender(effects, opts) {
```
Comment (ibgreen) [inline modules/core/src/lib/deck-renderer.js]: Nit: Type out `options`.

File: modules/core/src/passes/pick-layers-pass.js [hunk 3]
```diff
@@ -57,7 +56,8 @@ export default class PickLayersPass extends LayersPass {
         blend: !pickZ
       },
       () => {
-        this.drawLayers({
+        super.render({
```
Comment (ibgreen) [inline modules/core/src/passes/pick-layers-pass.js]: Nit: Any chance we can avoid introducing a call to `super`? What happens if we call `this.render()`? If we rename `super.render()`?

**Pull request merged**